src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -212,7 +238,7 @@ private function saveResource($href, Resource $resource, $returnType, array $que
                                           $href,
                                           json_encode($this->toStdClass($resource)),
                                           $query);
-
+        $this->removeResourceFromCache($resource);
```

Comment:
I think here is where we should be cacheing the response from create/save instead of from the other calling methods.